trunner/test/test_config.py

```diff
@@ -0,0 +1,187 @@
+import pytest
+import yaml
+
+from trunner.config import YAMLParser, YAMLParserError, TARGETS
+
+
+@pytest.fixture
+def parser(tmp_path, config, targets, request):
+    path = tmp_path / request.function.__name__
+    path = path.with_suffix('.yaml')
+    with open(path, 'w') as yaml_file:
+        yaml.dump(config, yaml_file)
+
+    return YAMLParser(path, targets)
+
+
+class TestTarget:
+    @pytest.fixture
+    def targets(self):
+        return TARGETS
+
+    @pytest.mark.parametrize('config, expect_target', [
+        (
```

Comment:
might have been easier/more readable to store raw yaml exernally, but I'm not against the current solution either.